Enable default bevy features in examples

SpecificProtagonist opened the pull request:
Fixes #15

Enable default bevy features (including x11) in examples/tests but not for using `bevy_magic_light_2d` as a dependency.

Also made Wayland optional. ~It's still enabled by default though because winit requires either it or X11.~

SpecificProtagonist committed:
enable default bevy features in examples

zicklag commented on Cargo.toml:
```diff
@@ -8,15 +8,15 @@ name = "bevy_magic_light_2d"
 crate-type = ["cdylib", "lib", "staticlib"]
 path = "src/lib.rs"
 
+[features]
+default = ["wayland"]
+wayland = ["bevy/wayland"]
+
```
I don't think these features should be added here.

Bevy plugins should allow the game crate to pick which bevy features it enables, without plugins enabling any of them by default, except for the ones require for the plugin to work.

It isn't necessary for the plugin to have a `wayland` feature because the game crate, which must depend on Bevy anyway, should be the one specifying that feature on Bevy.

dbenson24 replied:
I agree. Wayland should not be a feature

dbenson24 commented on Cargo.toml:
```diff
@@ -26,6 +26,9 @@ encase = "0.4.1"
 rand = "0.8.5"
 
+[dev-dependencies]
+bevy = { version = "0.9.1" }
```
I'm pretty sure that adding bevy with it's default features here is the only change that's needed.

SpecificProtagonist replied:
I also removed `bevy_asset`, `filesystem_watcher` and `png` because they were only needed for the examples and are now handled by dev-dependencies.

SpecificProtagonist commented:
I don't want wayland to be enabled by default either, I added the feature because otherwise cargo check fails. 
Removed it again; cargo check now needs to be run as `cargo check --features bevy/wayland` (or bevy/x11).

SpecificProtagonist commented:
While we're discussing `Cargo.toml`:

```toml
[profile.release]
codegen-units = 1
# debug = false
lto = "thin"
# opt-level = 3
panic = "abort"

[profile.dev.package."*"]
opt-level = 3
# debug = true
# incremental = true

[profile.dev]
codegen-units = 16
# opt-level = 0
# debug = true
# incremental = true
```

The commented lines can be removed as they match the default. `codegen-units = 16` can probably go to as the default is 256. Also, is the `panic = abort` and `codegen-units = 1` necessary? These profiles only affect examples, not programs that use bevy_magic_light_2d as a dependency.

zaycev reviewed (approved):
LGTM, thanks!
